src/generate/device.rs

```diff
@@ -100,21 +105,19 @@ pub fn render(d: &Device, target: &Target, nightly: bool, device_x: &mut String)
 
     out.extend(interrupt::render(target, &d.peripherals, device_x)?);
 
-    let core_peripherals: &[&str];
-
-    if fpu_present {
-        core_peripherals = &[
-            "CBP", "CPUID", "DCB", "DWT", "FPB", "FPU", "ITM", "MPU", "NVIC", "SCB", "SYST", "TPIU"
-        ];
+    let core_peripherals = if fpu_present {
+        vec![
```

Comment:
why did this become a vec? You can write, for example:

``` rust
let x: &[_] = if true {
    &[0, 1]
} else {
    &[2, 3, 4]
};
```

Reply:
Was recommended by clippy. Dunno. I can undo it.

Reply:
Oh, I remember. rustc complained about not being able to assign arrays with different lengths.

Reply:
TIL I can assign arrays of various lengths by declaring the type as `&[_]`. How odd. :-D